test/quantization/eager/test_numeric_suite_eager.py

```diff
@@ -534,3 +537,46 @@ def test_shadow_logger(self):
 
         self.assertEqual(len(logger.stats["float"]), 2)
         self.assertEqual(len(logger.stats["quantized"]), 2)
+
+    @skip_if_no_torchvision
+    def _test_vision_model(self, float_model):
+        float_model.to('cpu')
+        float_model.eval()
+        float_model.fuse_model()
+        float_model.qconfig = torch.quantization.default_qconfig
+        img_data = [(torch.rand(2, 3, 224, 224, dtype=torch.float), torch.randint(0, 1, (2,), dtype=torch.long)) for _ in range(2)]
+        qmodel = quantize(float_model, torch.quantization.default_eval_fn, [img_data], inplace=False)
+
+        wt_compare_dict = compare_weights(float_model.state_dict(), qmodel.state_dict())
+
+        def compute_error(x, y):
+            Ps = torch.norm(x)
+            Pn = torch.norm(x - y)
+            return 20 * torch.log10(Ps / Pn)
+
+        data = img_data[0][0]
+        # Take in floating point and quantized model as well as input data, and returns a dict, with keys
+        # corresponding to the quantized module names and each entry being a dictionary with two keys 'float' and
+        # 'quantized', containing the activations of floating point and quantized model at matching locations.
+        act_compare_dict = compare_model_outputs(float_model, qmodel, data)
+
+
+        for key in act_compare_dict:
+            compute_error(act_compare_dict[key]['float'][0], act_compare_dict[key]['quantized'][0].dequantize())
+
+        prepare_model_outputs(float_model, qmodel)
+
+        for data in img_data:
+            float_model(data[0])
+            qmodel(data[0])
+
+        # Find the matching activation between floating point and quantized modules, and return a dict with key
+        # corresponding to quantized module names and each entry being a dictionary with two keys 'float'
+        # and 'quantized', containing the matching floating point and quantized activations logged by the logger
+        act_compare_dict = get_matching_activations(float_model, qmodel)
+
+    @skip_if_no_torchvision
+    def test_mobilenet(self):
+        from torchvision.models.quantization import mobilenet_v2, mobilenet_v3_large
+        self._test_vision_model(mobilenet_v2(pretrained=True, quantize=False))
```

Comment:
can we do separate test cases for these two models, instead of combining into one test case?

Reply:
updated.